test/cpp/test_ir.cpp

```diff
@@ -279,5 +306,30 @@ TEST_F(IrTest, TestSizeDivNode) {
   });
 }
 
+TEST_F(IrTest, TestSizeDivNodeDynamic) {
+  int64_t num_non_zero_element = 1;
+  int64_t num_row = 10;
+  int64_t num_col = 10;
+  torch::lazy::NodePtr nonzero_node =
+      CreateNonZeroNode2d(num_non_zero_element, num_row, num_col);
+  torch::lazy::Value node_with_dynamism = torch::lazy::Value(nonzero_node, 0);
+
+  // static value = 100, dynamic value = 1
+  torch::lazy::NodePtr size_node_nonzero_0 =
+      torch::lazy::MakeNode<SizeNode>(node_with_dynamism, 0);
+  // static value = 2, dynamic value = 2
+  torch::lazy::NodePtr size_node_nonzero_1 =
+      torch::lazy::MakeNode<SizeNode>(node_with_dynamism, 1);
+
+  torch::lazy::NodePtr node_div = torch::lazy::MakeNode<SizeDiv>(
+      torch::lazy::Value(size_node_nonzero_0, 0),
+      torch::lazy::Value(size_node_nonzero_1, 0));
+
+  std::shared_ptr<torch::lazy::DimensionNode> dim_node_div =
+      std::dynamic_pointer_cast<torch::lazy::DimensionNode>(node_div);
+  EXPECT_EQ(dim_node_div->getStaticValue(), 50);
+  EXPECT_EQ(dim_node_div->getDynamicValue(), 0);
```

Comment:
This test doesn't check for divide by zero. How do we address that?

@vanbasten23

Reply:
Created a test: https://github.com/pytorch/xla/pull/4051